nltk/tag/__init__.py

```diff
@@ -114,6 +114,10 @@ def _pos_tag(tokens, tagset=None, tagger=None, lang=None):
             "Currently, NLTK pos_tag only supports English and Russian "
             "(i.e. lang='eng' or lang='rus')"
         )
+    # Throws Error if tokens is of string type
+    if str(tokens) == str:
```

Comment:
This `if` is inserted between another `if` and an `else`, which changes the control flow. In this case, however, both `if`s raise an uncaught exception (which exit the function immediately) so the behavior is the same. It would be better to use `elif`, particularly if we decide to issue a warning here instead of raise an error.

Also, for a type check it's clearer to check the type:

```python
elif isinstance(tokens, str):
    raise TypeError('tokens: expected a list of strings, got a string')
```

Reply:
I did refactor the changes, I believe it will output wrong result so it should be handled, maybe we can display a warning message when the passed parameter is a string.